lib/Bloc/exercise_bloc.dart

```diff
@@ -96,7 +96,7 @@ void deleteExercise(Exercise exercise) async {
   //checkRemoveMax(exercise);
   await firestoreInstance
       .collection("users")
-      .doc(userID)
+      .doc(SharedPref.getStringAsync(USER_UID))
```

Comment:
Since it is used multiple times here, we can declare a top-level String variable and store the value there and use it everywhere else in this file.